src/CMakeLists.txt

```diff
@@ -90,6 +92,11 @@ else()
     DEFAULT_LLVMSPIRV_BINARY_PATH="$<TARGET_FILE:llvm-spirv>")
 endif()
 
+if(MSVC)
+	target_compile_definitions(OpenCL-objects PRIVATE
+		NOMINMAX)
+endif()
```

Comment:
In windows.h (which is transitively included by some Vulkan or OpenCL headers) `min` and `max` are defined as macros -> messing with `std::min` and `std::max`.

Also pretty often for windows projects `UNICODE` is defined... but I'm not sure, whether this would make any difference in clvk, as it is not directly using any Win32 apis.. some [docs](https://docs.microsoft.com/en-us/cpp/text/support-for-unicode?view=vs-2019).

Reply:
Thanks for the explanation! Looking at the Vulkan headers I'm wondering whether this shouldn't be guarded by `WIN32` rather than `MSVC`.

Reply:
I agree :)